syntax = "proto3";

package SFG.SystemSimulator.NetworkMessages;

message RegisterRequest {
  string username = 1;
  string password_hash = 2;
}

message RegisterResponse {
  bool success = 1;
  string reason_for_fail = 2;
}

message LoginRequest {
  string username = 1;
  string password_hash = 2;
}

message LoginResponse {
  bool success = 1;
  string reason_for_fail = 2;
  string session_token = 3;
}

message CheckSessionRequest {
  string session_token = 1;
}

message CheckSessionResponse {
  bool is_valid = 1;
}

message LogoutRequest {
  string session_token = 1;
}

message LogoutResponse {
  bool success = 1;
  string reason_for_fail = 2;
}

message DeleteUserRequest {
  string username = 1;
  string password_hash = 2;
}

message DeleteUserResponse {
  bool success = 1;
  string reason_for_fail = 2;
}
